### PR TITLE
GroovyDoc and propertyMissing methods

### PR DESCRIPTION
**Pull request by SPalominos**
Add groovyDoc in OSMTools and implements propertyMissing methods to simplify the methods call.

### CODE REVIEW

File: osm/src/main/groovy/org/orbisgis/osm/OSMTools.groovy [hunk 1]
```diff
@@ -8,38 +8,53 @@ import org.slf4j.LoggerFactory
...
+    /** {@link Loader} script with its {@link Process}. */
     public static Loader = new Loader()
+    /** {@link Transform} script with its {@link Process}. */
     public static Transform = new Transform()
+    /** {@link Utilities} script with its {@link Process}. */
     public static Utilities = new Utilities()
 
-    //Utility methods
-    static def getUuid() { UUID.randomUUID().toString().replaceAll("-", "_") }
-    static def uuid = { UUID.randomUUID().toString().replaceAll("-", "_") }
```
Comment (ebocher) [inline osm/src/main/groovy/org/orbisgis/osm/OSMTools.groovy:27]: Why changing this name ?

File: osm/src/main/groovy/org/orbisgis/osm/Transform.groovy [hunk 2]
```diff
@@ -844,7 +844,7 @@ static boolean extractNodesAsPoints(JdbcDataSource datasource, String osmTablesP
             ${epsgCode}) as the_geom ${createTagList(datasource, columnsSelector)} from ${osmTablesPrefix}_node as a, 
           ${osmTablesPrefix}_node_tag  b where a.id_node=b.id_node group by a.id_node;"""
         }else {
-            def FILTERED_NODES = "FILTERED_NODES_$uuid"
+            def FILTERED_NODES = "FILTERED_NODES_${uuidCl()}"
```
Comment (ebocher) [inline osm/src/main/groovy/org/orbisgis/osm/Transform.groovy]: use OSMTools.uuid

File: osm/src/main/groovy/org/orbisgis/osm/OSMTools.groovy (hunk 1 above)
Comment (SPalominos) [reply osm/src/main/groovy/org/orbisgis/osm/OSMTools.groovy:27]: We had two declaration of the 'uuid' method:
 - `def uuid = {...}` was for a normal calling (outside a String) by using `uuid()`
 - `def getUuid() {...}` was for calling inside a String by using `"$uuid`

Now with the propertyMissing method, there is only one implementation which works everywheren and which is more compliant with the groovy syntax/mecanism

File: osm/src/main/groovy/org/orbisgis/osm/Transform.groovy (hunk 2 above)
Comment (SPalominos) [reply osm/src/main/groovy/org/orbisgis/osm/Transform.groovy]: You're right !

File: osm/src/main/groovy/org/orbisgis/osm/OSMTools.groovy (hunk 1 above)
Comment (SPalominos) [reply osm/src/main/groovy/org/orbisgis/osm/OSMTools.groovy:27]: Fix in a better way with just `static def getUuid(){...}`.

**Pull request merged**